src/v/cluster/partition_leaders_table.cc

```diff
@@ -88,29 +97,51 @@ void partition_leaders_table::update_partition_leader(
         auto [new_it, _] = _leaders.emplace(
           leader_key{
             model::topic_namespace(ntp.ns, ntp.tp.topic), ntp.tp.partition},
-          leader_meta{.current_leader = leader_id, .update_term = term});
+          leader_meta{
+            .current_leader = leader_id,
+            .update_term = term,
+            .partition_revision = revision_id});
         it = new_it;
     } else {
+        // Currently we have to check if revision id is valid since not all the
+        // code paths devlivers revision information
+        //
+        // TODO: always check revision after we will add revision to metadata
+        // dissemination requests
+        const bool revision_id_valid = revision_id >= model::revision_id{0};
+        // skip update for partition with previous revision
+        if (revision_id_valid && revision_id < it->second.partition_revision) {
```

Comment:
is the case handled here where `revision_id` is being compared to `it->second.partition_revision` and `it->second.partition_revision` was initialized with `model::revision_id{}`. I guess in that case revision_id >= 0 and thus won't be < the `model::revision{}` value so we'll pass through.

Reply:
this is never the case , partition revisionis based on the `_raft0` log offset offset when preset is always defined